abad.go

```diff
@@ -97,4 +121,103 @@ func (a *Abad) evalUnaryExpr(expr *ast.UnaryExpr) (Obj, error) {
 	}
 
 	return num, nil
+}
+
+func (a *Abad) evalExpr(n ast.Node) (types.Value, error) {
+	if !ast.IsExpr(n) {
+		panic("internal error: not an expression")
+	}
+
+	switch n.Type() {
+	case ast.NodeNumber:
+		val := n.(ast.Number)
+		return types.Number(val.Value()), nil
+	case ast.NodeIdent:
+		val := n.(ast.Ident)
+		return a.evalIdentExpr(val)
+	case ast.NodeMemberExpr:
+		val := n.(*ast.MemberExpr)
+		return a.evalMemberExpr(val)
+	case ast.NodeCallExpr:
+		val := n.(*ast.CallExpr)
+		return a.evalCallExpr(val)
+	case ast.NodeUnaryExpr:
+		expr := n.(*ast.UnaryExpr)
+		return a.evalUnaryExpr(expr)
+	}
+
+	panic("unreachable")
+	return nil, nil
+}
+
+func (a *Abad) evalIdentExpr(ident ast.Ident) (types.Value, error) {
+	val, err := a.global.Get(utf16.Str(ident))
+	if err != nil {
+		return nil, err
+	}
+
+	if types.StrictEqual(val, types.Undefined) {
```

Comment:
I may be wrong on this, but isn't undefined evaluateable ? Like if Im evaluating an identifier and it is undefined the expression should return undefined (like null), no ? I think I can even sum undefineds in JS =P

Reply:
not in 'strict mode'. As we'd discussed in the past, I've implemented just the 'strict' path. As far as I remember the project we're aiming uses 'strict mode'.

Reply:
![js](https://user-images.githubusercontent.com/251478/41792454-4a05ad96-762f-11e8-931b-9742bcad7b8d.PNG)

Reply:
Did not know that, thanks =D (I agree with being strict, I thought it would also work on strict mode)